rfhub2/db/repository/collection_repository.py

```diff
@@ -14,9 +16,41 @@ class CollectionRepository(IdEntityRepository):
     def _items(self) -> Query:
         return self.session.query(Collection).options(selectinload(Collection.keywords))
 
+    @property
+    def _items_with_stats(self) -> Query:
+        collection_statistics = (
+            self.session.query(
+                (func.sum(Statistics.times_used)).label("times_used"),
+                Statistics.collection,
+            )
+            .group_by(Statistics.collection)
+            .subquery()
+        )
+        return (
+            self.session.query(Collection, collection_statistics.c.times_used)
+            .outerjoin(
+                collection_statistics,
+                Collection.name == collection_statistics.c.collection,
+            )
+            .options(selectinload(Collection.keywords))
+        )
+
     def _id_filter(self, item_id: int) -> BinaryExpression:
         return Collection.id == item_id
 
+    @staticmethod
+    def filter_criteria(pattern: Optional[str], libtype: Optional[str]):
+        filter_criteria = []
+        if pattern:
+            filter_criteria.append(Collection.name.ilike(glob_to_sql(pattern)))
+        if libtype:
+            filter_criteria.append(Collection.type.ilike(glob_to_sql(libtype)))
+        return filter_criteria
+
+    @staticmethod
+    def from_stats_row(row: Tuple[Collection, int]) -> CollectionWithStats:
```

Comment:
I ws roaming around that solution and the one above, but gave up.  Now it's easy to understand how it work and how to tackle this in future. Awesome! :)